src/test/ai-accuracy-tests/ai-accuracy-tests.ts

```diff
@@ -0,0 +1,349 @@
+
+/**
+ * Insert the generative ai results to a db
+ * so we can track how they perform overtime.
+ */
+async function pushResultsToDB({
+  results,
+  anyFailedAccuracyThreshold,
+  runTimeMS,
+  httpErrors,
+}: {
+  results: TestResult[];
+  anyFailedAccuracyThreshold: boolean;
+  runTimeMS: number;
+  httpErrors: number;
+}) {
+  const client = new MongoClient(
+    process.env.AI_ACCURACY_RESULTS_MONGODB_CONNECTION_STRING || ''
+  );
+
+  try {
+    const database = client.db(TEST_RESULTS_DB);
+    const collection = database.collection(TEST_RESULTS_COL);
+
+    const gitCommitHash = await execFile('git', ['rev-parse', 'HEAD'], {
+      cwd: projectRoot,
+    });
+
+    const doc = {
+      gitHash: gitCommitHash.stdout.trim(),
+      completedAt: new Date(),
+      attemptsPerTest: ATTEMPTS_PER_TEST,
+      anyFailedAccuracyThreshold,
+      httpErrors,
+      totalRunTimeMS: runTimeMS, // Total elapsed time including timeouts to avoid rate limit.
+      results: results.map((result) => {
+        const { 'Avg Execution Time (ms)': runTimeMS, Pass, ...rest } = result;
+        return {
+          runTimeMS,
+          Pass: Pass === '✓',
+          ...rest,
+        };
+      }),
+    };
+
+    await collection.insertOne(doc);
+  } finally {
+    await client.close();
+  }
+}
+
+const buildMessages = (testCase: TestCase) => {
+  switch (testCase.type) {
+    case 'generic':
+      return GenericPrompt.buildMessages({
+        request: { prompt: testCase.userInput },
+        context: { history: [] },
+      });
+
+    case 'query':
+      return QueryPrompt.buildMessages({
+        request: { prompt: testCase.userInput },
+        context: { history: [] },
+        databaseName: testCase.databaseName,
+        collectionName: testCase.collectionName,
+      });
+
+    case 'namespace':
+      return NamespacePrompt.buildMessages({
+        request: { prompt: testCase.userInput },
+        context: { history: [] },
+      });
+
+    default:
+      throw new Error(`Unknown test case type: ${testCase.type}`);
+  }
+};
+
+async function runTest({
+  testCase,
+  aiBackend,
+}: {
+  testCase: TestCase;
+  aiBackend: AIBackend;
+}) {
+  const messages = buildMessages(testCase);
+  const chatCompletion = await aiBackend.runAIChatCompletionGeneration({
+    messages: messages.map((message) => ({
+      ...message,
+      role:
+        message.role === vscode.LanguageModelChatMessageRole.User
+          ? 'user'
+          : 'assistant',
+    })),
+  });
+
+  return chatCompletion;
+}
+
+describe('AI Accuracy Tests', function () {
+  let cluster: MongoCluster;
+  let mongoClient: MongoClient;
+  let fixtures: Fixtures = {};
+  let anyFailedAccuracyThreshold = false;
+  let startTime;
+  let aiBackend;
+  let connectionString: string;
+
+  const results: TestResult[] = [];
+  const testOutputs: TestOutputs = {};
+
+  this.timeout(60_000 /* 1 min */);
+
+  before(async function () {
+    console.log('Starting setup for AI accuracy tests...');
```

Comment:
Can we use maybe `ora` here so GitHub doesn't complain about console logs?

Reply:
Oh it's complaining since I'm missing a `eslint-disable no-console` - thanks ! Adding now.